### PR TITLE
Add types to express all known attribute value types.

### PR DESCRIPTION
**Pull request by geeknoid**

### CODE REVIEW

File: mixer/v1/config/cfg.proto [hunk 1]
```diff
@@ -129,3 +129,23 @@ message ClientConfig {
...
   repeated AspectRule rules = 3;
 }
+
+// Uri represents a properly formed URI.
+message Uri {
+  string uri = 1;
+}
+
+// IPAddr holds an IPv4 or IPv6 address.
+message IPAddr {
```
Comment (wora) [inline mixer/v1/config/cfg.proto]: IpAddress

File: mixer/v1/config/cfg.proto [hunk 2]
```diff
@@ -129,3 +129,23 @@ message ClientConfig {
...
+  bytes ip_addr = 1;
+}
+
+// DNSNames holds a valid domain name.
+message DNSName {
```
Comment (wora) [inline mixer/v1/config/cfg.proto]: DnsName

File: mixer/v1/config/cfg.proto [hunk 3]
```diff
@@ -129,3 +129,23 @@ message ClientConfig {
...
+  string dns_name = 1;
+}
+
+// EmailAddr holds a properly formatted email address.
+message EmailAddr {
```
Comment (wora) [inline mixer/v1/config/cfg.proto]: EmailAddress

File: mixer/v1/config/cfg.proto [hunk 4]
```diff
@@ -129,3 +129,23 @@ message ClientConfig {
   string revision = 2;
   repeated AspectRule rules = 3;
 }
+
+// Uri represents a properly formed URI.
+message Uri {
```
Comment (wora) [inline mixer/v1/config/cfg.proto]: If we use Camel case for Uri, we should do the same for all names.

File: mixer/v1/config/cfg.proto [hunk 5]
```diff
@@ -129,3 +129,23 @@ message ClientConfig {
...
+}
+
+// DNSNames holds a valid domain name.
+message DNSName {
+  string dns_name = 1;
+}
+
+// EmailAddr holds a properly formatted email address.
+message EmailAddr {
+  string email_addr = 1;
```
Comment (wora) [inline mixer/v1/config/cfg.proto]: For all well known types under google/protobuf, we use generic "value" inside the message. I suggest we do the same here.

**Comment by geeknoid** [comment]
Updated per Hong's comments.

**Pull request merged**